src/network/NetworkManagerPrimary.cc

```diff
@@ -129,19 +142,23 @@ bool NetworkManagerPrimary::Step(const UpdateInfo &_info)
     return false;
   }
 
-  private_msgs::SimulationStep step;
-  step.mutable_stats()->CopyFrom(convert<msgs::WorldStatistics>(_info));
-
-  // Affinities that changed this step
-  this->PopulateAffinities(step);
-
   // Check all secondaries are ready to receive steps - only do this once at
   // startup
   if (!this->SecondariesCanStep())
   {
     return false;
   }
 
+  // Different step method for render networks
+  if (this->IsRenderNetwork())
+    return this->RenderingStep(_info);
+
+  private_msgs::SimulationStep step;
+  step.mutable_stats()->CopyFrom(convert<msgs::WorldStatistics>(_info));
+
+  // Affinities that changed this step
+  this->PopulateAffinities(step);
```

Comment:
I think this needs to be done before the `SecondariesCanStep` so we don't miss it. If we leave it for later, we may not know anymore that the affinities changed (maybe, I haven't checked).